react-native/react-native-flipper/react_native_flipper.rb

```diff
@@ -0,0 +1,44 @@
+def flipper_post_install(installer)
+  installer.pods_project.targets.each do |target|
+    if target.name == 'YogaKit'
+      target.build_configurations.each do |config|
+        config.build_settings['SWIFT_VERSION'] = '4.1'
+      end
+    end
+  end
+end
+
+
+def use_flipper!(versions = {}, configurations: ['Debug'])
```

Comment:
Why do we have to define this ? Shouldn't  this come from RN core ? Are you suggesting  to move this configuration in Flipper ? cc  @mweststrate

Reply:
Yes, currently it's coming from RN core.
my suggestion is to move this to Flipper to be able to change this without needing to upgrade inside the RN core

Reply:
Then I have few questions. How will we ensure that bumping up the deps in this function won't break RN master and the scheduled release of RN, thats why I feel its better that this configuration is part of RN core rather than here. 

Library/Framework releases shouldn't be worried about the clients integrating and if a client builds(ofcourse we have our own flipper CI in place to avoid breakages) or not, its rather clients  responsibility to verify if they are compatible with latest library/framework release version and its them who should take the call to bump rather than us. Isn't it ?

@mweststrate  what do you think ?

Reply:
Yes, you are correct, but in this PR I'm implementing on `react-native-flipper` suggesting to this library to be an integration with Flipper. The idea is this dep should take care of how to make it work with RN.

Since `react-native-flipper` is an integration with custom Flipper Plugins, I think it should be able to handle not only custom plugins but the Flipper integration itself. It ma be an more easy way to manage Flipper issues on RN side.

The plan if the PR is merged is to change the default template to include the `react-native-flipper` dep by default using the default initialization methods that are already in this dependency, but it can be customized on project side (like the current RN template already does)

Reply:
@b3coded I think it is a good idea to first raise this as a proposal discussion (RFC) on the react-native repo itself, so that folks over there are involved as well. I'm still a bit unclear what problems this solves:

1. customizing the flipper version, or removing it, is all done in the podfile of your repository with the `use_flipper` command. I'm not sure how that is made easier over here?
2. by no longer being part of the RN core setup, signal on issues will arrive much later


So, before diving into the technical correctness of the solution / code, I'd love to better understand the proposal better first? You state  "This takes advantage of the react-native-flipper dependency to make possible more easy way to install, remove and update Flipper without needing to change the react-native repo.", but I'm missing the how / why of that statement :). Could you show how a typical react native project looks before and after the changes proposed? And how the impact of bumping a Flipper version is different? And how the compatibility between Flipper and RN would still be guaranteed by CI? This might be a very good proposal, but so far I don't see enough rationale yet :).